client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -541,12 +541,41 @@ class CalypsoifyIframe extends Component< Props & ConnectedProps & ProtectedForm
 			: `Block Editor > ${ postTypeText } > New`;
 	};
 
-	onIframeLoaded = ( iframeUrl: string ) => {
-		if ( ! this.successfulIframeLoad ) {
-			window.location.replace( iframeUrl );
+	onIframeLoaded = async ( iframeUrl: string ) => {
+		// Shortcut to avoid polling mechanism if not needed.
+		if ( this.successfulIframeLoad ) {
```

Comment:
I tested to make sure that it's possible to remove this shortcut, and the behavior makes sense. In that case, the interval only runs once and the correct finish condition is run. The cancelPolling mechanism is working, since the interval didn't run any more times than it had to 👍